pkg/registry/node/strategy.go

```diff
@@ -153,17 +153,30 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
-		// Ignore requested scheme, use scheme provided by GetConnectionInfo
+	// We check if we want to get a default Kubelet's transport. It happens if either:
+	// - no port is specified in request (Kubelet's port is default),
+	// - we're using Port stored as a DaemonEndpoint and requested port is a Kubelet's port stored in the DaemonEndpoint,
+	// - there's no information in the API about DaemonEnpoint (legacy cluster) and requested port is equal to ports.KubeletPort (cluster-wide config)
+	if portReq == "" ||
+		(node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 &&
+			strconv.Itoa(node.Status.DaemonEndpoints.KubeletEndpoint.Port) == portReq) ||
+		(node.Status.DaemonEndpoints.KubeletEndpoint.Port == 0 &&
+			strconv.Itoa(ports.KubeletPort) == portReq) {
```

Comment:
I think this logic can be made a lot more understandable if you first figure out what the default port is by checking node.Status.DaemonEndpoints.KubeletEndpoint.Port and ports.KubeletPort?